server/src/main/java/com/linecorp/centraldogma/server/internal/storage/repository/cache/CachingRepository.java

```diff
@@ -113,10 +106,40 @@ public CompletableFuture<Map<String, Entry<?>>> find(Revision revision, String p
         requireNonNull(options, "options");
 
         final Revision normalizedRevision = normalizeNow(revision);
-        return cache.get(new CacheableFindCall(repo, normalizedRevision, pathPattern, options))
-                    .handleAsync((unused, cause) -> {
+
+        Map<FindOption<?>, ?> cacheableOptions = options;
+        final Integer maxEntries = (Integer) options.get(FindOption.MAX_ENTRIES);
+        if (maxEntries != null) {
+            final ImmutableMap.Builder<FindOption<?>, Object> newOptions = ImmutableMap.builder();
+            options.forEach((key, value) -> {
+                if (key != FindOption.MAX_ENTRIES) {
+                    newOptions.put(key, value);
+                }
+            });
+            cacheableOptions = newOptions.build();
+        }
+
+        return cache.get(new CacheableFindCall(repo, normalizedRevision, ALL_PATH, cacheableOptions))
```

Comment:
I understood worst case, the cached size can be `(number of revisions) * (size of directory)`.

Reply:
Possible. Based on the user pattern I observed, most clients were only interested in HEAD revision. Old revisions were accessed occasionally. The practical cached size would be `(1 or 2 revisions) * (size of directory)`.